components/core/tests/test-EncodedVariableInterpreter.cpp

```diff
@@ -237,25 +224,37 @@ TEST_CASE("EncodedVariableInterpreter", "[EncodedVariableInterpreter]") {
         // Open writer
         VariableDictionaryWriter var_dict_writer;
         var_dict_writer.open(cVarDictPath, cVarSegmentIndexPath,
-                             EncodedVariableInterpreter::get_var_dict_id_range_end() - EncodedVariableInterpreter::get_var_dict_id_range_begin());
+                             EncodedVariableInterpreter::get_var_dict_id_max());
 
         // Test encoding
         vector<encoded_variable_t> encoded_vars;
         vector<variable_dictionary_id_t> var_ids;
-        vector<string> var_strs = {"4938", to_string(EncodedVariableInterpreter::get_var_dict_id_range_begin()),
-                                   "-25.5196868642755", "-00.00", "bin/python2.7.3"};
-        msg = "here is a string with a small int " + var_strs[0] + " and a very large int " + var_strs[1] + " and a double " + var_strs[2] +
-              " and a weird double " + var_strs[3] + " and a str with numbers " + var_strs[4];
+
+        string large_val_str = to_string(EncodedVariableInterpreter::get_var_dict_id_max()) + "0";
+        vector<string> var_strs = {"4938", large_val_str, "-25.5196868642755",
+                                   "-00.00", "bin/python2.7.3"};
+        msg = "here is a string with a small int " + var_strs[0] +
+                " and a very large int " + var_strs[1] +
+                " and a double " + var_strs[2] +
+                " and a weird double " + var_strs[3] +
+                " and a str with numbers " + var_strs[4];
+
         LogTypeDictionaryEntry logtype_dict_entry;
-        EncodedVariableInterpreter::encode_and_add_to_dictionary(msg, logtype_dict_entry, var_dict_writer, encoded_vars, var_ids);
+        EncodedVariableInterpreter::encode_and_add_to_dictionary(msg, logtype_dict_entry,
+                                                                 var_dict_writer, encoded_vars,
+                                                                 var_ids);
         var_dict_writer.close();
 
         // Test var_ids is correctly populated
         size_t encoded_var_id_ix = 0;
-        for (const auto& var : encoded_vars) {
-            if(EncodedVariableInterpreter::is_var_dict_id(var)){
+        LogTypeDictionaryEntry::VarDelim var_placeholder;
+        for (auto var_ix = 0; var_ix < logtype_dict_entry.get_num_vars(); var_ix++) {
+            static_cast<void>(logtype_dict_entry.get_var_info(var_ix, var_placeholder));
```

Comment:
Instead of using `static_cast<void>` to explicitly ignore the return, I think assigning `std::ignore = ...` is more functional, especially if we mark `get_var_info` as `[[nodiscard]]`.